sdk/internal/recording/matchers_test.go

```diff
@@ -0,0 +1,97 @@
+//go:build go1.16
+// +build go1.16
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package recording
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestAddBodilessMatcher(t *testing.T) {
+	err := Start(t, packagePath, nil)
+	require.NoError(t, err)
+
+	err = AddBodilessMatcher(t, nil)
+	require.NoError(t, err)
+
+	err = AddBodyRegexSanitizer("*", "", nil)
+	require.NoError(t, err)
+
+	req, err := http.NewRequest("POST", "https://localhost:5001", nil)
+	require.NoError(t, err)
+
+	req.Header.Set(UpstreamURIHeader, "https://bing.com")
+	req.Header.Set(ModeHeader, GetRecordMode())
+	req.Header.Set(IDHeader, GetRecordingId(t))
+
+	client, err := GetHTTPClient(t)
+	require.NoError(t, err)
+
+	_, err = client.Do(req)
+	require.NoError(t, err)
+
+	err = Stop(t, nil)
+	require.NoError(t, err)
+
+	jsonFile, err := os.Open(fmt.Sprintf("./testdata/recordings/%s.json", t.Name()))
+	require.NoError(t, err)
+	defer jsonFile.Close()
+
+	var data RecordingFileStruct
+	byteValue, err := ioutil.ReadAll(jsonFile)
+	require.NoError(t, err)
+	err = json.Unmarshal(byteValue, &data)
+	require.NoError(t, err)
+	require.Equal(t, data.Entries[0].RequestBody, "")
```

Comment:
I feel like I'm missing something here; doesn't this just verify the body regex sanitizer worked? Don't you need to send a second request, identical to the first save for its content, to verify the bodiless matcher is set?

Reply:
The CI runs in playback so that's sort of the check here. However, you're correct this would be a more thorough way to test the matcher works.